### ISSUE TITLE
Use case for non-CERN users + Google Cal

### ISSUE THREAD

**Issue by cha0sg0d** [indico/newdle#362]
Hi all,

I'm a big fan of CERN (thank you Sir Tim BL!). I've been searching everywhere for open source software that provides 1-click meeting scheduling and integrates with existing calendars like Outlook and Gcal. 

I was wondering if someone could point me in the right direction to repurpose this codebase for a non-CERN audience and gogole calendar. Thanks!

**Comment by ThiefMaster** [comment]
Hi, integration with Google Calendar is probably not trivial, but if someone contributed this it'd be amazing of course.

Probably you would need to use Google OIDC/OAuth for the login, store the token, and then use the Google Calendar APIs for free-busy information.

**Comment by cha0sg0d** [comment]
Yeah def not trivial. I have experience with the OAuth and API calls, but less so with the actual scheduling algorithm. If the correct free busy info is gathered, will Newdle calculate the times? Or does it just ping Microsoft's algorithm.

**Comment by ThiefMaster** [comment]
I would expect Google to have an API that returns free-busy information in a decent format. You can check the existing free-busy sources [here](https://github.com/indico/newdle/tree/master/newdle/providers/free_busy) to see how newdle needs the data.

There are no scheduling algorithms involved. People specify their preferred time slots and the (human) newdle creator then picks the best slot based on the users' answers.

**Comment by cha0sg0d** [comment]
Oh nice. Ok, I'll play around with it and PR when I can.

**Comment by amCap1712** [comment]
I implemented this feature but their is one main blocker, the auth system currently relies on the auth token being stored on the frontend. AFAIU, no user login data is currently stored in the session/cookies. For accessing a user's google calendar, we first need to have them authenticate their Google Account with newdle using an OAuth 2 flow (different from the multipass login). 

During this flow, when the user is redirected after granting authorization to newdle the server is unable to determine the current logged in user because the request comes from google's servers and not newdle frontend and thus lacks an authorization header.

I guess I can as a special case store the Participant.auth_uid in the session before triggering the auth flow but I am curious if there's a special reason to avoid the use of session/cookies for login management here.
